com.unity.render-pipelines.universal/Editor/Converter/ReadonlyMaterialConverter.cs

```diff
@@ -0,0 +1,279 @@
+{
+    enum IdentifierType { kNullIdentifier = 0, kImportedAsset = 1, kSceneObject = 2, kSourceAsset = 3, kBuiltInAsset = 4 };
+
+    internal static class ReadonlyMaterialMap
+    {
+        public static readonly Dictionary<string, string> Map = new Dictionary<string, string>
+        {
+            {"Default-Diffuse", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/Lit.mat"},
+            {"Default-Material", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/Lit.mat"},
+            {"Default-ParticleSystem", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/ParticlesUnlit.mat"},
+            {"Default-Particle", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/ParticlesUnlit.mat"},
+            {"Default-Terrain-Diffuse", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/TerrainLit.mat"},
+            {"Default-Terrain-Specular", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/TerrainLit.mat"},
+            {"Default-Terrain-Standard", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/TerrainLit.mat"},
+            {"Sprites-Default", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/Sprite-Unlit-Default.mat"},
+            {"Sprites-Mask", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/Sprite-Unlit-Default.mat"},
+
+            // These are custom URP shaders right now, but the old BiRP ones do render in Unity. Do we need to convert?
+            {"SpatialMappingOcclusion", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/SpatialMappingOcclusion.mat"},
+            {"SpatialMappingWireframe", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/SpatialMappingWireframe.mat"},
+
+            // TODO: Replace these with something more appropriate
+            {"Default UI Material", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/Lit.mat"},
+            {"ETC1 Supported UI Material", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/Lit.mat"},
+            {"Default-Line", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/Lit.mat"},
+            {"Default-Skybox", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/Lit.mat"},
+        };
+    }
+
+    internal class ReadonlyMaterialConverter : RenderPipelineConverter
+    {
+        public override string name => "Readonly Material Converter";
+        public override string info => "Converts references to Built-In readonly materials to URP readonly materials. This will create a .index file and that can take a long time.";
+        public override Type container => typeof(BuiltInToURPConverterContainer);
+        public override bool needsIndexing => true;
+
+        public override void OnInitialize(InitializeConverterContext ctx, Action callback)
+        {
+            using (var context = Search.SearchService.CreateContext("asset", "urp:convert-readonly"))
```

Comment:
Since you are handling the search context asynchronously you probably do not want to have the search context be disposed when it goes out of scope in the caller code because of the `using`?

Consider removing the `using` and calling `context?.Dispose()` after `callback.Invoke();`?

Reply:
Done